File: java/cloud-based-color-grading/common/src/main/java/cn/gsjgsj/common/entity/Role.java
package cn.gsjgsj.common.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 角色实体类
 */
@Data
@TableName("sys_role")
public class Role {
    
    /**
     * 角色ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 角色名称
     */
    private String roleName;
    
    /**
     * 角色标识（如ROLE_ADMIN）
     */
    private String roleKey;
    
    /**
     * 描述
     */
    private String description;
    
    /**
     * 状态：0-正常 1-禁用
     */
    private Integer status;
    
    /**
     * 创建时间
     */
    private LocalDateTime createTime;
}

File: java/cloud-based-color-grading/management-vue-project/src/types/role.ts
/**
 * 角色相关类型定义
 */

// 角色
export interface Role {
  id: number
  roleName: string
  roleKey: string
  description: string
  status: number              // 0-正常 1-禁用
  createTime: string
}

// 角色表单数据
export interface RoleFormData {
  roleName: string
  roleKey: string
  description: string
  status: number
}
